src/store/modules/models/helpers.js

```diff
@@ -195,5 +195,427 @@ const helpers = {
         return state.library[type];
     }
   },
+
+  /**
+   * Generates a hash containing all the space types for the given building type
+   *
+   * @param {string} building_type Type of building to generate space types for
+   * @param {string} template Template that the space is following
+   * @param {boolean} wholeBuilding Whether or not the space types are generated for the whole building
+   * @returns {{[key: string]: string}}
+   */
+  spaceTypesForBuilding(building_type, template, wholeBuilding = false) {
```

Comment:
This logic is pulled from https://github.com/NREL/openstudio-extension-gem/blob/develop/lib/openstudio/extension/core/os_lib_model_generation.rb#L404-L696.  If I know for sure that we don't need the additional data there,  I can simplify the logic here and cut down on some of the redundancy.

Reply:
it looks like the keys and values are always identical. Maybe a Set or Array would be a better choice of data structure?

Reply:
The UI expects to be returned an object with id - name pairings; however, I forgot to update the names to be user friendly, though, so I'll go through and update them.